### PR TITLE
fix #954, ubuntu, diskfull, enable the pre-script log and post-script log can be forwarded to the MN

### LINKED ISSUE #954: [FVT]Failed to find expected logs while site.xcatdebugmode=1 for ubuntu on ppc64le
Description:
According to the design doc:　https://github.com/xcat2/xcat-core/wiki/support%20new%20values%20for%20site.xcatdebugmode
while the site.xcatdebugmod=1/2 for ubuntu installation, the Y3: installation logs can be forwarded to the MN in /var/log/xcat/computes.log file without export to subsequent commands.
But there isn't this kind of file in MN:/var/log/xcat, only the below two logs:
```
root@c910f03fsp03v10:/var/log/xcat# ls
commands.log  xcat.log
```
And failed to find most of the excepted logs in the two files.
ENV.
```
root@c910f03fsp03v10:/opt/xcat/share/xcat/tools/autotest# lsxcatd -a
Version 2.12 (git commit 641af3c6a0fcec382c4482d8c0abc6717c06af3d, built Thu Apr 14 09:32:00 EDT 2016)
This is a Management Node
dbengine=SQLite
```

### PR DESCRIPTION
**Pull request by penguhyang**
fix #954 
For ubuntu diskfull installation, enable the pre-script log and post-script log can be forwarded to the MN.

### CODE REVIEW

**Comment by whowutwut** [comment]
In the description, you mention pre and post script log can be forwarded to the mgmt node, but the files affected is only pre.*, is something missing? 

Also some of the code change is for adding `xcatdebugmode` code processing, that is missed in the description 

Shouldn't we have these file specify the ubuntu version? ubuntu16, ubuntu14? what OS/ARCH is are using the pre.ubuntu file?

File: xCAT-server/share/xcat/install/scripts/pre.ubuntu [hunk 1]
```diff
@@ -11,6 +11,24 @@ fi
 
 debconf-get open-iscsi/targets > /tmp/q
 
+cat >/tmp/syslogd.sh <<EOF
```
Comment (whowutwut) [inline xCAT-server/share/xcat/install/scripts/pre.ubuntu]: Suggest that we specify a filename with xcat in it... `/tmp/xcat.syslogd.sh`

File: xCAT-server/share/xcat/install/scripts/pre.ubuntu.ppc64 [hunk 2]
```diff
@@ -6,6 +11,24 @@ fi
 
 debconf-get open-iscsi/targets > /tmp/q
 
+cat >/tmp/syslogd.sh <<EOF
```
Comment (whowutwut) [inline xCAT-server/share/xcat/install/scripts/pre.ubuntu.ppc64]: Suggest that we specify a filename with xcat in it.., something like: `/tmp/xcat.syslogd.sh`

**Comment by penguhyang** [comment]
@whowutwut Thanks for review.
I create a new process to connect master using busybox syslogd when running pre-script file. So the pre script(run in installer) and post script(run in chroot /target) log can be forwarded to the mgmt node. The config file created in /tmp... it's only used in installer time.

File: xCAT-server/share/xcat/install/scripts/pre.ubuntu (hunk 1 above)
Comment (penguhyang) [inline xCAT-server/share/xcat/install/scripts/pre.ubuntu]: fixed.

File: xCAT-server/share/xcat/install/scripts/pre.ubuntu.ppc64 (hunk 2 above)
Comment (penguhyang) [inline xCAT-server/share/xcat/install/scripts/pre.ubuntu.ppc64]: fixed.

**Comment by immarvin** [comment]
inside the installer, there is already a syslogd running, which write all the logs into /var/log/syslog:
````
/sbin/syslogd -m 0 -O /var/log/syslog -S
````
we should 1st kill the existing syslogd process  first and then start the syslogd with remote log forwarding.

the following lines can be added into the beginning of "/xCAT-server/share/xcat/install/scripts/pre.ubuntu" and "/xCAT-server/share/xcat/install/scripts/pre.ubuntu.ppc64"

````
if [ "$XCATDEBUGMODE" -gt "0" ]; then
    killall syslogd
    syslogd -R #XCATVAR:XCATMASTER#
fi
````
btw, the documentation on busybox is https://busybox.net/downloads/BusyBox.html

**Comment by penguhyang** [comment]
@immarvin Thanks for review.
Code updated.
1. 1st kill the existing syslogd process
2. use syslogd without create a new file.

**Comment by immarvin** [comment]
agree to merge

**Pull request merged**